docs/src/modules/java/pages/actions-as-controller.adoc

```diff
@@ -3,86 +3,176 @@
+
+In this case we mark that the entity has been created using a creation timestamp in the shopping cart state stored on first
+`create` call (when the timestamp has the default value of `0`). If the cart has already been stored with a timestamp we return an error effect:
+
+[.tabset]
+Java::
++
+.src/main/java/com/example/shoppingcart/domain/ShoppingCart.java
+[source,java]
 ----
-include::example$scala-valueentity-counter/src/main/scala/com/example/actions/DoubleCounterAction.scala[tag=controller-forward]
+include::example$java-valueentity-shopping-cart/src/main/java/com/example/shoppingcart/domain/ShoppingCart.java[tag=create]
 ----
-<1> On incoming requests, we double the value of `IncreaseValue`
-<2> We call `components.counter.increase()` to create a `DeferredCall` that is then passed to `effects.forward()` method.
 
-== Registering the Action
+Scala::
++
+.src/main/java/com/example/shoppingcart/domain/ShoppingCart.scala
+[source,scala]
+----
+include::example$scala-valueentity-shopping-cart/src/main/scala/com/example/shoppingcart/domain/ShoppingCart.scala[tag=create]
+----
 
-To make Akka Serverless aware of the Action, we need to register it with the service.
 
-From the code-generation, the registration gets automatically inserted in the generated `AkkaServerlessFactory.withComponents` method from the `Main` class.
+== Composing calls
 
-When we add an Action to an existing service, like we did here, the `Main` class needs to be adapted. In this example, we build on top
-of the  xref:value-entity.adoc[Value Entity Counter] example. Therefore, the generated `AkkaServerlessFactory.withComponents` is now accepting
-two registrations: the `Counter` and the `DoubleCounterAction`, and we need to adapt it.
+The async call shown in the previous section, can also be used to chain or compose multiple calls to a single action response.
+
+In this example we build on the previous cart creation by adding an initial item in the cart once it has been created,
+but before we return the new id to the client:
 
 [.tabset]
 Java::
 +
+.src/main/java/com/example/shoppingcart/ShoppingCartActionImpl.java
 [source,java]
-./src/main/java/com/example/Main.java
 ----
-include::example$java-valueentity-counter/src/main/java/com/example/Main.java[tag=registration]
+include::example$java-valueentity-shopping-cart/src/main/java/com/example/shoppingcart/ShoppingCartActionImpl.java[tag=createPrePopulated]
 ----
+<1> `CompletionStage#thenCompose` allow us to perform an additional async operation, returning a `CompletionStage` once the current one completes successfully
+<2> Create a request to add an initial item to the cart
+<3> Executing the `addItem` call returns a `CompletionStage<Empty>` once it succeeds
+<4> `handle` allows us to transform the successful completion of `addItem` with `Empty` to the response type of this method - `NewCartCreated`
+<5> `effects().asyncReply()` lets us reply once the `CompletionStage<NewCartCreated>` completes.
```

Comment:
tiny thing, but be consistent with no period at the end of the callout comments?

```suggestion
<5> `effects().asyncReply()` lets us reply once the `CompletionStage<NewCartCreated>` completes
```